### ISSUE TITLE
Issue in convert to utc

### ISSUE THREAD

**Issue by dileepkumars1** [nodatime/nodatime#968]
Hi,

I got an issue in the version 1.3.2 which is fixed in the latest stable version. The issue is, when we call convert to utc on "Europe/London" 29-10-2017 01:00 - 01:55 (DST end time is 02:00), it is not correct. The issue is fixed in the current release. Can I have the defect number for that as I have to give my customer a reference.

Thanks much,
Dileep

**Comment by jskeet** [comment]
It's hard to tell exactly what you mean in your description. If you could provide a small code sample it would be much easier to help you.

**Comment by dileepkumars1** [comment]
Hi Jon,

Sorry for my vague description. I am adding some screen shots with the code and issue. I am using the nodatime version 1.3.2 with tzdb2017 database.

![image](https://user-images.githubusercontent.com/29368827/29922276-8cbbc936-8e72-11e7-9b30-042bd978ee7c.png)

and the output is 

![image](https://user-images.githubusercontent.com/29368827/29922295-a43b0de2-8e72-11e7-93fc-2582f17e6ed1.png)

When I upgraded the nodatime version to 2.2.0, the output is corrected and looks like

![image](https://user-images.githubusercontent.com/29368827/29922512-92063614-8e73-11e7-9eed-65d284378b13.png)

The code I use to convert to UTC is as follows

![image](https://user-images.githubusercontent.com/29368827/29922570-cd3e07fc-8e73-11e7-8131-e5304ce2a85c.png)

My question is do you have an identified defect regarding this. I am asking this because it is fixed in the current version. I need to provide my customer a defect/reference number so that they can also refer the issue.

Thanks much,
Dileep

**Comment by jskeet** [comment]
For future reference, it's *much* more helpful to include both code and output as text rather than screenshots. It would also help if you didn't show examples using types that aren't part of Noda Time - just a complete example showing the use of `DateTimeZone` would be fine.

However, it looks like this isn't a bug at all - it's just a change to what `LocalDateTime.InZoneLeniently` does. Both are behaving as documented.

[1.3 documentation](http://nodatime.org/1.3.x/api/NodaTime.LocalDateTime.html#NodaTime_LocalDateTime_InZoneLeniently_NodaTime_DateTimeZone_):

> Returns the mapping of this local date/time within the given Date​Time​Zone, with "lenient" rules applied such that ambiguous values map to the later of the alternatives, and "skipped" values map to the start of the zone interval after the "gap".

[2.0 documentation](http://nodatime.org/2.0.x/api/NodaTime.LocalDateTime.html#NodaTime_LocalDateTime_InZoneLeniently_NodaTime_DateTimeZone_): 

> Returns the mapping of this local date/time within the given Date​Time​Zone, with "lenient" rules applied such that ambiguous values map to the earlier of the alternatives, and "skipped" values are shifted forward by the duration of the "gap".

To achieve the 2.0 behaviour, you just need to use `LocalDateTime.InZone` passing in a `ZoneMappingResolver`, which you'd create with `Resolvers.CreateMappingResolver(Resolvers.ReturnEarlier, Resolvers.ReturnStartOfIntervalAfter)`.

That would only change the behaviour for *ambiguous* mappings - if you wanted the matching behaviour for *skipped* mappings, you'd need to basically copy the 2.0 code for [`ReturnForwardShifted`](https://github.com/nodatime/nodatime/blob/master/src/NodaTime/TimeZones/Resolvers.cs#L87), but that's easy enough to do.

Closing this issue as the code is behaving exactly as documented - there's no "identified defect"; we changed the lenient behaviour to return earlier rather than return later, but neither is always "correct".
